my_pyscf/mcpdft/__init__.py

```diff
@@ -64,16 +64,43 @@ def _laspdftEnergy(mc_class, mc_or_mf_or_mol, ot, ncas_sub, nelecas_sub, DoLASSI
 
     if frozen is not None: mc1 = mc_class(mf_or_mol, ncas_sub, nelecas_sub, ncore=ncore, spin_sub = spin_sub, frozen=frozen)
     else:  mc1 = mc_class(mf_or_mol, ncas_sub, nelecas_sub, ncore=ncore, spin_sub = spin_sub)
-    
+
     from mrh.my_pyscf.mcpdft.laspdft import get_mcpdft_child_class
     mc2 = get_mcpdft_child_class(mc1, ot, DoLASSI=DoLASSI, **kwargs)
 
     if mc0 is not None:
-        mc2.mo_coeff = mc_or_mf_or_mol.mo_coeff.copy ()    
+        mc2.mo_coeff = mc_or_mf_or_mol.mo_coeff.copy ()
         mc2.ci = copy.deepcopy (mc_or_mf_or_mol.ci)
         mc2.converged = mc0.converged
     return mc2
- 
+
+
+def _lassipdftEnergy(mc_class, mc_or_mf_or_mol, ot, ncas_sub, nelecas_sub, DoLASSI=False, ncore=None, spin_sub=None,
+                   frozen=None, **kwargs):
+
+    from mrh.my_pyscf.lassi import lassi, lassis
+
+    if isinstance(mc_or_mf_or_mol, (lassi.LASSI, lassis.LASSIS)):
```

Comment:
It is unnecessary to check both lassi.LASSI and lassis.LASSIS: the latter is a child class of the former, so you only need to check the former.

Reply:
Done.